### PR TITLE
Accessors for differential and algebraic erquations

### PR DESCRIPTION
**Pull request by TorkelE**
Creates 10 different general purpose accessors for handling differential and algebraic functions within systems:
```
is_alg_equation
is_diff_equation
alg_equations
diff_equations
has_alg_equations
has_diff_equations
get_alg_eqs
get_diff_eqs
has_alg_eqs
has_diff_eqs
```
Examples:
```julia
using ModelingToolkit
using ModelingToolkit: t_nounits as t, D_nounits as D
@parameters p d
@variables X(t)
eq1 = D(X) ~ p - d*X
eq2 = 0 ~ p - d*X
@named osys1 = ODESystem([eq1], t)
@named osys2 = ODESystem([eq2], t)
osys12 = compose(osys1, [osys2])
osys21 = compose(osys2, [osys1])

is_alg_equation(eq1) # returns `false`.
is_alg_equation(eq2) # returns `true`.
is_diff_equation(eq1) # returns `true`.
is_diff_equation(eq2) # returns `false`.

has_alg_equations(osys1) # returns `false`.
has_alg_equations(osys2) # returns `true`.
has_diff_equations(osys1) # returns `true`.
has_diff_equations(osys2) # returns `false`.

alg_equations(osys1) # returns `Equation[]`.
alg_equations(osys2) # returns `[ 0 ~ p - d*X(t)]`.
diff_equations(osys1) # returns `[Differential(t)(X(t)) ~ p - d*X(t)]`.
diff_equations(osys2) # returns `Equation[]`.

get_alg_eqs(osys12) # returns `Equation[]`.
get_alg_eqs(osys21) # returns `[0 ~ p - d*X(t)]`.
get_diff_eqs(osys12) # returns `Differential(t)(X(t)) ~ p - d*X(t)`.
get_diff_eqs(osys21) # returns `Equation[]`.

has_alg_eqs(osys12) # returns `true`.
has_alg_eqs(osys21) # returns `false`.
has_diff_eqs(osys12) # returns `true`.
has_diff_eqs(osys21) # returns `false`.
```

There are docstrings and tests for all functions.

These are needed in Catalyst v14 to handle equations when you convert to `ODESystem`s that may (or may not) contain algebraic equations. Sincet hey are general-purpose, I figured it was better to put them here than in Catalyst.

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Format and document

**Comment by TorkelE** [comment]
Got docstrings, so can link them in the docs, where would you want them? In the long term (as part of this PR or not), would it be useful to have an "API" page on the docs which just lists functions like these and their docstrings?

**Comment by ChrisRackauckas** [comment]
No, API dumps are always bad. Make a proper manual page on equations. Add the `has_algebraic` and such to the right system docs pages (ODESystem, SDESystem, etc.)

**Comment by TorkelE** [comment]
Ok, I have added docs and run the formater.

File: docs/src/systems/ODESystem.md [hunk 1]
```diff
@@ -13,8 +13,16 @@ ODESystem
...
+  - `continuous_events(sys)`: The set of continuous events in the ODE.
+  - `discrete_events(sys)`: The set of discrete events in the ODE.
+  - `alg_equations(sys)`: The algebraic equations (i.e. that does not contain a differential) that defines the ODE.
+  - `get_alg_eqs(sys)`: The algebraic equations (i.e. that does not contain a differential) that defines the ODE. Only returns equations of the current-level system.
+  - `diff_equations(sys)`: The differential equations (i.e. that contain a differential) that defines the ODE.
+  - `get_diff_eqs(sys)`: The differential equations (i.e. that contain a differential) that defines the ODE. Only returns equations of the current-level system.
+  - `has_alg_equations(sys)`: Returns `true` if the ODE contains any algebraic equations (i.e. that does not contain a differential).
+  - `has_alg_eqs(sys)`: Returns `true` if the ODE contains any algebraic equations (i.e. that does not contain a differential). Only considers the current-level system.
+  - `has_diff_equations(sys)`: Returns `true` if the ODE contains any differential equations (i.e. that does contain a differential).
```
Comment (ChrisRackauckas) [inline docs/src/systems/ODESystem.md:24]: We should use the new non-canonical, but we can handle that in a future PR

Comment (TorkelE) [reply docs/src/systems/ODESystem.md:24]: Sounds good

Comment (TorkelE) [reply docs/src/systems/ODESystem.md:24]: thanks for helping getting this one sorted out quickly!

**Pull request merged**